src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props

```diff
@@ -31,6 +31,11 @@ Copyright (c) .NET Foundation. All rights reserved.
     <None Remove="**/*$(DefaultLanguageSourceExtension)" Condition=" '$(EnableDefaultCompileItems)' == 'true' "/>
     <None Remove="**/*.resx" Condition=" '$(EnableDefaultEmbeddedResourceItems)' == 'true' "/>
   </ItemGroup>
+  <ItemGroup Condition=" '$(EnableDefaultDependentUponMetadata)' == 'true' And '$(EnableDefaultResxDependentUponMetadata)' == 'true' ">
+    <Compile Update="**/*.Designer.cs" DependentUpon="$([System.IO.Path]::ChangeExtension($([System.IO.Path]::GetFileNameWithoutExtension('%(Identity)')), '.resx'))" />
```

Comment:
Ideally, we would condition this on whether or not a matching resx file exists...

However, MSBuild does not let you condition on the Identity item metadata outside of a target and CPS doesn't run targets for the tree view (at least that I could find/remember).

Reply:
Not knowing if the target exists seems like a blocker to me. There are things other than .resx that can have .Designer.cs. What happens if we bring up WinForms et al? How will we know whether the .Designer.cs should match resx or something else?

Reply:
Right. This might need a project-system or MSBuild feature (https://github.com/Microsoft/msbuild/issues/2346) to get it working properly.

Reply:
Actually, it looks like @dsplaisted commented that this works if we move it to proper metadata rather than the "inline metadata": https://github.com/Microsoft/msbuild/issues/2346#issuecomment-332095422

Reply:
Fixed.